projects/packages/forms/src/dashboard/inbox/bulk-actions-menu.js

```diff
@@ -1,34 +1,72 @@
-import { AntiSpamIcon } from '@automattic/jetpack-components';
-import { DropdownMenu, MenuGroup, MenuItem } from '@wordpress/components';
+import { Button } from '@wordpress/components';
+import { useDispatch, useSelect } from '@wordpress/data';
 import { __ } from '@wordpress/i18n';
-import { chevronDown, trash, archive, inbox } from '@wordpress/icons';
+import { updateResponseStatus } from '../data/responses';
+import { STORE_NAME } from '../state';
+
+const ACTIONS = {
+	moveToTrash: 'trash',
+	removeFromTrash: 'untrash',
+	markAsSpam: 'mark_as_spam',
+	markAsNotSpam: 'mark_as_not_spam',
+	spamCheck: 'spam_check',
+};
+
+const VIEWS = {
+	inbox: 'inbox',
+	spam: 'spam',
+	trash: 'trash',
+};
+
+const ActionsMenu = ( { currentView, selectedResponses, setSelectedResponses } ) => {
+	const { fetchResponses, setLoading } = useDispatch( STORE_NAME );
+	const query = useSelect( select => select( STORE_NAME ).getResponsesQuery(), [] );
+
+	const onActionHandler = action => async () => {
+		try {
+			setLoading( true );
+			await updateResponseStatus( selectedResponses, action );
+			fetchResponses( query );
+			setSelectedResponses( [] );
+		} catch {
+			//TODO: Implement error handling
+		}
+	};
 
-const ActionsMenu = () => {
 	return (
-		<DropdownMenu
-			label={ __( 'Bulk actions', 'jetpack-forms' ) }
-			text={ __( 'Bulk actions', 'jetpack-forms' ) }
-			icon={ chevronDown }
-			popoverProps={ { placement: 'bottom-end' } }
-			toggleProps={ { variant: 'secondary', iconPosition: 'right' } }
-		>
-			{ ( { onClose } ) => (
-				<MenuGroup>
-					<MenuItem onClick={ onClose } icon={ <AntiSpamIcon size="22" /> } iconPosition="left">
-						{ __( 'Spam check', 'jetpack-forms' ) }
-					</MenuItem>
-					<MenuItem icon={ inbox } iconPosition="left">
-						{ __( 'Mark unread', 'jetpack-forms' ) }
-					</MenuItem>
-					<MenuItem icon={ archive } iconPosition="left">
-						{ __( 'Archive', 'jetpack-forms' ) }
-					</MenuItem>
-					<MenuItem icon={ trash } iconPosition="left">
-						{ __( 'Delete', 'jetpack-forms' ) }
-					</MenuItem>
-				</MenuGroup>
+		<>
+			{ currentView !== VIEWS.trash && (
+				<Button onClick={ onActionHandler( ACTIONS.moveToTrash ) } variant="secondary">
```

Comment:
Hmm, should we `disable` these buttons while `loading` is `true`? That should prevent any accidental race conditions with multiple requests going out simultaneously as well as providing some initial UI feedback to the user.

Reply:
The buttons are not visible while `loading` is `true`.
I'm hiding the Bulk Actions menu completely here: https://github.com/Automattic/jetpack/pull/29766/files#diff-f6dedbab4016eb9af19b412e0c0be98ae4342e84c9c69aa770554719c50983b7R80